src/compat/posix/fnmatch/fnmatch.c

```diff
@@ -0,0 +1,148 @@
+/**
+ * @file
+ * @brief fnmatch_function
+ *
+ * @date 20.12.18
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <errno.h>
+#include <fnmatch.h>
+
+static int fnmatch_string(const char *pattern, const char *string, int flags) {
+	int flag = 0;
+	int denial_flag = 0;
+	while (*pattern != '\0') {
+		flag = 0;
+		denial_flag = 0;
+
+		if ((*string == '\0') && (*pattern != '*')) {
+			return FNM_NOMATCH;
+		}
+		switch (*pattern) {
+
+			case '?':
+				break;  
+
+			case '*':
+				pattern++;
+				if (*pattern == '\0') {
+					return 0;
+				}
+				while (*string != '\0') {
+					if (!fnmatch_string(pattern,string,flags)) {
+						return 0;
+					}
+					string++;
+				}
+				return FNM_NOMATCH;
+
+			case '/':
+				if (*pattern != *string) {
+					return FNM_NOMATCH;
+				}
+				if (flags == FNM_PATHNAME) {
+					return 0;
+				}
+				break;		
+
+			case '[':
+				pattern++;
+				if (*pattern == '^') {
+					denial_flag = 1;
+				}
+				while (*pattern != ']') {
+					if ((denial_flag) && (*string == *pattern)) {
```

Comment:
You can make it even shorter, like

```
while(...) {
if (*string == *pattern) { flag = 1; break; }
}

if (flag && !denial) || (!flag && denial) {return FNM_NOMATCH}
```
```

Reply:
btw this place is wrong cuz in FNM_PATHNAME pattern '[a/b]' mean '[a' + '/' + 'b]' so i need to rework this place

Reply:
reworked and made shorter